✨ feat: Add offline cache for lookup ens (1 week)

insulineru opened the pull request.

insulineru committed:
♻️ refactor: singleton rpc provider

insulineru committed:
🔊 chore: add some logs for debug

insulineru committed:
♻️ refactor: remove circular dependency

insulineru committed:
1.1.8-rc.0

insulineru committed:
♻️ refactor: prevent duplicate auth logic (with logs)

insulineru committed:
1.1.8-rc.1

insulineru committed:
Update QueryProvider to add defaultOptions with cacheTime and staleTime

insulineru committed:
Fix ENS lookup caching bug in getDomainAddress function of evm.ts

insulineru committed:
Add @tanstack/query-sync-storage-persister and @tanstack/react-query-persist-client to package.json

insulineru committed:
Update @tanstack/react-query from v4.1.3 to v4.24.10

insulineru committed:
Commit: Add local storage persister to QueryProvider.tsx

insulineru committed:
1.1.8-rc.2

insulineru committed:
Update persister key for web3-wallets.store in localStoragePersister

insulineru committed:
1.1.8-rc.3

iower commented on src/utils/evm.ts:
```diff
@@ -29,5 +36,6 @@ export const isEvmContract = async (chainId: number, address: string) => {
   const { rpc_url: rpc } = getNetworkById(chainId)
   const provider = new ethers.providers.JsonRpcProvider(rpc)
   const addressCode = await provider.getCode(address)
+  console.log('[isEvmContract]')
```
```suggestion
```

iower reviewed (approved):
Looks OK

nrjshka commented on src/constants.ts:
```diff
@@ -138,3 +139,5 @@ export const ERC20_GAS_LIMIT = 300000
 
 export const LOCAL_STORAGE_WALLETS_ADDRESSES = 'web3-wallets-addresses'
+
+export const EVM_PROVIDER = new ethers.providers.JsonRpcProvider('https://rpc.ankr.com/eth')
```
`https://rpc.ankr.com/eth` to const

nrjshka replied:
and use const to initialize `EVM_PROVIDER`

nrjshka commented on src/context/QueryProvider.tsx:
```diff
@@ -1,16 +1,32 @@
 }
 
-const queryClient = /* #__PURE__ */ new QueryClient()
+const queryClient = /* #__PURE__ */ new QueryClient({
+  defaultOptions: {
+    queries: {
+      cacheTime: 1000 * 60 * 60 * 24 * 7, // 1 week
+      staleTime: 1000 * 60 * 60 * 2 // 2 hours
+    }
+  },
+})
```
We use `react-query` in `useBalances` as well. With such settings it might be affected. Have you checked it?

insulineru replied:
All balances queries has own `refetchInterval`

insulineru committed:
🚑 fix: Replace EVM_PROVIDER with ETHEREUM_PROVIDER in related files

insulineru committed:
🔧 chore: update package version to 1.1.8

This commit updates the package version from 1.1.8-rc.3 to 1.1.8 in the package.json file.
